report.go

```diff
@@ -54,23 +116,44 @@ func writeInvocation(w io.Writer) {
 		if current < line-contextLines {
 			continue
 		}
-		prefix := fmt.Sprintf("        %d", current)
+		linePrefix := fmt.Sprintf("%s%d", prefix, current)
 		if current == line {
 			found = true
-			prefix += "!"
+			linePrefix += "!"
 		}
-		fmt.Fprintln(tw, prefix+"\t"+sc.Text())
+		fmt.Fprint(tw, prefixf(linePrefix+"\t", "%s", sc.Text()))
 	}
 	tw.Flush()
 	if err = sc.Err(); err != nil {
-		fmt.Fprintf(w, "<cannot scan source file: %s>\n", err)
+		fmt.Fprintf(w, prefixf(prefix, "<cannot scan source file: %s>", err))
 		return
 	}
 	if !found {
-		fmt.Fprintln(w, "<cannot find source lines>")
+		fmt.Fprintf(w, prefixf(prefix, "<cannot find source lines>"))
 	}
 }
 
-// contextLines holds the number of lines of code to show when showing a
-// failure context.
-const contextLines = 3
+// prefixf formats the given string with the given args. It also inserts the
+// final newline if needed and indentation with the given prefix.
+func prefixf(prefix, format string, args ...interface{}) string {
+	var buf bytes.Buffer
+	lines := strings.Split(fmt.Sprintf(format, args...), "\n")
```

Comment:
var buf []byte
	s := strings.TrimSuffix(fmt.Sprintf(format, args...), "\n")
	for _, line := range strings.Split(s, "\n") {
	    buf = append(buf, prefix...)
	    buf = append(buf, line...)
	    buf = append(buf, '\n')
	}
	return string(buf)

?

Reply:
Done.